docs/storageclass-description.md

```diff
@@ -0,0 +1,126 @@
+This document describes the design of the StorageClass parameters for the
+file-based Gluster CSI driver, `org.gluster.glusterfs`. It is intended to
+describe the desired interface for configuring Gluster-file based
+persistent volume types.
+
+This document presents a number of example StorageClass definitions, followed
+by a full yaml description at the end of the document.
+
+# Examples
+
+## Simple replica 3
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: simpleR3
+provisioner: org.gluster.glusterfs
+parameters:
+  volumeType:
+    type: replicate
+    replicate:
+      replicas: 3
+```
+
+## Replica 3 with arbiter
+
+This creates a volume with 2 data nodes and a 3rd brick that contains just file
+metadata (the arbiter brick). This example also shows setting volume options.
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: r3Arbiter
+provisioner: org.gluster.glusterfs
+parameters:
+  volumeType:
+    type: replicate
+    replicate:
+      replicas: 3
+      arbiterType: normal
+  volumeOptions:
+    "performance.cache-refresh-timeout": "60"
+    "performance.cache-size": "134217728"
+    "performance.nl-cache": "on"
+    "performance.md-cache-timeout": "300"
+```
+
+## Erasure coded volume
+
+This defines a template for 4+2 erasure coded volumes. Additionally, the bricks
+will be spread across three specific availability zones. (Gluster nodes are
+tagged with the AZ in which they reside).
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: simpleR3
+provisioner: org.gluster.glusterfs
+parameters:
+  volumeType:
+    type: disperse
+    disperse:
+      data: 4
+      redundancy: 2
+  volumeLayout:
+    dataZones:
+      - rack1
+      - rack2
+      - rack7
+```
+
+# Full yaml description
+
+Below is the full description of the StorageClass, complete with all optional
+fields.
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: reliable
+provisioner: org.gluster.glusterfs
+parameters:
+  # Maximum size of an individual brick. Volumes that exceed this will be
+  # created as distributed-* volumes.
+  maxBrickSize: 100Gi
```

Comment:
Is this actually a feature of Gluster I'm not aware of? Or is this logic yet-to-be-implemented in the CSI driver?

Reply:
This will interact w/ GD2 IVP to limit brick size to make recovery and capacity balancing easier. I *thought* Heketi had something similar, though not quite used like intended here... Perhaps I'm just thinking of vol resize.

Reply:
Ah, okay. Might be something that could be enforced within the driver, for Heketi anyway.